sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -805,33 +799,34 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
       cause = null)
   }
 
-  def fileNotFoundError(e: FileNotFoundException): SparkFileNotFoundException = {
-    new SparkFileNotFoundException(
-      errorClass = "_LEGACY_ERROR_TEMP_2062",
-      messageParameters = Map("message" -> e.getMessage))
+  def fileNotExistError(path: String, e: Exception): Throwable = {
+    new SparkException(
```

Comment:
Why did you change `SparkFileNotFoundException` to `SparkException`. This can break user's apps, I think. If they catch the exact `SparkFileNotFoundException`, they will miss your `SparkException`. Especially, after you changed the error class too. How should user know about your changes?

Reply:
Note: at the end, we threw `SparkException` before: https://github.com/apache/spark/pull/44953/files#diff-158a89983d40e2ebba7048dd10bac07b6c4fa03d70ee608923b093b314219f19L254 . This change is to make the behavior closer to before.